### PR TITLE
✨ Implement new cbBTC/USDC Merit Action aprs

### PR DESCRIPTION
**Pull request by MartinGbz**
## General Changes

The new "Supply cbBTC / Borrow USDC" action is coming very soon to Merit.
This PR implement the necessary refactoring to accept any Merit actions (other than just stkGHO APR currently).

Recap:
- refactored the `MeritIncentivesButton` & `useMeritIncentives`
- merge the logic of getting specific user apr & global apr to only 1 hook and component
- create a unique source of data (`symbolToMeritData`) for Merit incentives, in order to easily add other Merit incentives in the future.
- Include aUSDC to IncentivesSymbolMap

## Developer Notes

For now the API return is mocked because the route is not yet available. So you can now test these different cases:
- no wallet connected => use the average Merit APR (should show an APR of 8%)
- wallet connected but user not eligible for merit => use the average Merit APR (should show an API of 5% (5% was just to test to verify we pass in this case)
- wallet connected and eligible to Merit => use the specific user Merit APR (should show an API of 10%)

I'll update the PR to remove the mock once the new version of the API will be available.

Reminder: there are differences in APRs between users on Merit due to the boosters system. Here in this action, users who have migrated from awBTC to acbBTC or from vUSDT to vUSDC, get a boost, so they have a higher APR.

---

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)

### CODE REVIEW

**Comment by foodaka** [comment]
Screenshot 2024-10-21 at 20 44 30
Lets make this wrap similar to other incentives (see metis)

File: src/components/incentives/IncentivesButton.tsx [hunk 1]
```diff
@@ -230,6 +195,7 @@ const Content = ({
           {incentives.length < 5 ? (
             <>
               {incentives.map(getSymbolMap).map((incentive) => {
+                console.log('### incentive', incentive);
```
Comment (foodaka) [inline src/components/incentives/IncentivesButton.tsx:198]: when its ready just remove the debug logs

File: src/components/incentives/MeritIncentivesTooltipContent.tsx [hunk 2]
```diff
@@ -6,13 +6,17 @@ import { FormattedNumber } from '../primitives/FormattedNumber';
 import { Link } from '../primitives/Link';
 import { Row } from '../primitives/Row';
 import { TokenIcon } from '../primitives/TokenIcon';
+import { getSymbolMap } from './IncentivesTooltipContent';
 
 export const MeritIncentivesTooltipContent = ({
-  incentiveAPR,
-  rewardTokenSymbol,
-}: Omit<ReserveIncentiveResponse, 'rewardTokenAddress'>) => {
+  meritIncentives,
```
Comment (foodaka) [inline src/components/incentives/MeritIncentivesTooltipContent.tsx:12]: what is this type?

File: src/hooks/useMeritIncentives.ts [hunk 3]
```diff
@@ -2,46 +2,138 @@ import { ReserveIncentiveResponse } from '@aave/math-utils/dist/esm/formatters/i
...
+      let meritIncentives: MeritIncentives | null = null;
+
+      if (user) {
+        console.log('USER APRS');
+        // const response = await fetch(`${url}?user=${user}`);
+        // const data = await response.json();
+        // meritIncentives = data.currentAPR as MeritIncentives;
+        let mockedAprs;
+        if (user == '0xf8E0D20E5548f3b607547ACec4149Ef9d951Df73'.toLowerCase()) {
+          mockedAprs = {
```
Comment (foodaka) [inline src/hooks/useMeritIncentives.ts:84]: remember to clean this up

File: src/components/incentives/IncentivesButton.tsx [hunk 4]
```diff
@@ -3,9 +3,10 @@ import { ReserveIncentiveResponse } from '@aave/math-utils/dist/esm/formatters/i
 import { DotsHorizontalIcon } from '@heroicons/react/solid';
 import { Box, SvgIcon, Typography } from '@mui/material';
 import { useState } from 'react';
-import { useMeritIncentives, useUserMeritIncentives } from 'src/hooks/useMeritIncentives';
+import { useMeritIncentives } from 'src/hooks/useMeritIncentives';
 import { useRootStore } from 'src/store/root';
 import { DASHBOARD } from 'src/utils/mixPanelEvents';
+import { Side } from 'src/utils/utils';
```
Comment (foodaka) [inline src/components/incentives/IncentivesButton.tsx:9]: Lets use the `ProtocolAction` for this. 

```
import {

  ProtocolAction,
} from '@aave/contract-helpers';
ProtocolAction.borrow
```

File: src/hooks/useMeritIncentives.ts [hunk 5]
```diff
@@ -2,46 +2,138 @@ import { ReserveIncentiveResponse } from '@aave/math-utils/dist/esm/formatters/i
...
+        meritIncentives = mockedAprs;
+      }
+
+      return meritIncentives;
     },
     queryKey: ['meritIncentives'],
     staleTime: 1000 * 60 * 5,
     select: (data) => {
-      // rewards are always in GHO, for now
+      const meritReserveIncentiveData = symbolToMeritData(symbol, market);
```
Comment (foodaka) [inline src/hooks/useMeritIncentives.ts:123]: How about something like

```
      const meritData = getMeritData(symbol, market);
      
      const getMeritData = (symbol: string, market: string): MeritReserveIncentiveData | undefined => 
  MERIT_DATA_MAP[market]?.[symbol];
  
  const MERIT_DATA_MAP: Record<string, Record<string, MeritReserveIncentiveData>> = {
  [CustomMarket.proto_mainnet_v3]: {
    GHO: {
      action: MeritAction.ETHEREUM_STKGHO,
      rewardTokenAddress: AaveV3Ethereum.ASSETS.GHO.UNDERLYING,
      rewardTokenSymbol: 'GHO',
    },
    cbBTC: {
      action: MeritAction.SUPPLY_CBBTC_BORROW_USDC,
      rewardTokenAddress: AaveV3Ethereum.ASSETS.USDC.A_TOKEN,
      rewardTokenSymbol: 'aEthUSDC',
      side: Side.SUPPLY,
    },
    USDC: {
      action: MeritAction.SUPPLY_CBBTC_BORROW_USDC,
      rewardTokenAddress: AaveV3Ethereum.ASSETS.USDC.A_TOKEN,
      rewardTokenSymbol: 'aEthUSDC',
      side: Side.BORROW,
    },
  },
};

```

Comment (foodaka) [reply src/hooks/useMeritIncentives.ts:123]: instead of that complex switch

**Comment by foodaka** [comment]
Screenshot 2024-10-22 at 10 41 34
seems issue here on staking page

**Comment by MartinGbz** [comment]
> 0xf8E0D20E5548f3b607547ACec4149Ef9d951Df73

done ✅

**Comment by MartinGbz** [comment]
> Screenshot 2024-10-22 at 10 41 34 seems issue here on staking page

Ah weird?
Don't have the issue on my side (even with eligible/non-eligible/disconnected users)
I'll dig this

Capture d’écran 2024-10-22 à 12 04 38

File: src/components/incentives/MeritIncentivesTooltipContent.tsx (hunk 2 above)
Comment (MartinGbz) [reply src/components/incentives/MeritIncentivesTooltipContent.tsx:12]: It's of type `ReserveIncentiveResponse` why?

File: src/components/incentives/IncentivesButton.tsx (hunk 4 above)
Comment (MartinGbz) [reply src/components/incentives/IncentivesButton.tsx:9]: Ah good catch, I used the "Side" type that was implement on the "Superfest" tooltip PR.
I'll replace all Side by ProtocolAction, perfect

File: src/hooks/useMeritIncentives.ts (hunk 5 above)
Comment (MartinGbz) [reply src/hooks/useMeritIncentives.ts:121]: Way better like this I agree

**Comment by MartinGbz** [comment]
> > Screenshot 2024-10-22 at 10 41 34 seems issue here on staking page
> 
> Ah weird? Don't have the issue on my side (even with eligible/non-eligible/disconnected users) I'll dig this
> 
> Capture d’écran 2024-10-22 à 12 04 38

Okay it seems it was an issue regarding the mock I did. I fixed it, but it will be cleaner when the API will be ready

**Review by grothem — APPROVED** [review]

**Review by foodaka — APPROVED** [review]

**Pull request merged**